### PR TITLE
TH-1022 | Add scroll restoration to collection page

### PR DESCRIPTION
**Pull request by Sartonon**
## Description :sparkles:

- Add scroll restoration to collection page

## Additional notes :spiral_notepad:

I spent quite a bit of time investigating how scroll positions could be handled and if we could have some reusable logic here but ended up with simple solution for now... If someone has good suggestions how to could be handled lets hear them :)

### CODE REVIEW

**Comment by Sartonon** [comment]
I spent quite a bit of time investigating how scroll positions could be handled and if we could have some reusable logic here but ended up with simple solution for now... If someone has good suggestions how to could be handled lets hear them :)

File: src/domain/collection/CollectionPageContainer.tsx [hunk 1]
```diff
@@ -82,6 +85,24 @@ const CollectionPageContainer: React.FC = () => {
...
+
+  // Scroll back to correct event card (event tat user is coming back from)
+  React.useEffect(() => {
+    if (location.state?.eventId) {
+      const cardId = getLargeEventCardId(location.state.eventId);
+      scrollToEventCard(cardId);
+      // Clear eventId value to keep scroll position correctly
+      const state = { ...location.state };
+      delete state.eventId;
+      history.replace({ ...location, state });
```
Comment (Sartonon) [inline src/domain/collection/CollectionPageContainer.tsx:102]: Not sure in which situation this is really needed, it is same as in EventSearchPageContainer.tsx

Comment (nikomakela) [reply src/domain/collection/CollectionPageContainer.tsx:102]: Found this with a quick search: https://stackoverflow.com/a/47513369/784642

**Review by nikomakela — APPROVED** [review]
Looks good to me.

**Comment by nikomakela** [comment]
Actually, I tested it from PR server and it seems to not work on my Chrome Version 88.0.4324.192 (Official Build) (x86_64). 

I clicked "Kolmen karhun..." 
image

I get this 
image

After browser's back button, I'm on top of the page!

**Comment by nikomakela** [comment]
Okay, by clicking left-arrow -button it seems to work. 
image

File: src/domain/collection/CollectionPageContainer.tsx (hunk 1 above)
Comment (Sartonon) [reply src/domain/collection/CollectionPageContainer.tsx:102]: I was just thinking in what situation this could cause problems if left out...

Comment (Sartonon) [reply src/domain/collection/CollectionPageContainer.tsx:102]: Without this if I would navigate back and forward I would still get restored scroll, not sure if that would be problematic

**Pull request merged**